### PR TITLE
pass prometheus route prefix to thanos sidecar

### LINKED ISSUE #2344: Thanos sidecar does not get configured properly when using routePrefix
**What did you do?**
Deployed prometheus with a routePrefix option

**What did you expect to see?**
Thanos query sidecar get updated with the proper prometheus-url argument.

**What did you see instead? Under which circumstances?**
Thanos sidecar did not respect the prometheus route prefix and was thus unable to function as expected

### PR DESCRIPTION
**Pull request by ephur**
Fixes #2344

### CODE REVIEW

File: pkg/prometheus/statefulset.go [hunk 1]
```diff
@@ -658,7 +658,7 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 
 		thanosArgs := []string{
 			"sidecar",
-			fmt.Sprintf("--prometheus.url=http://%s:9090", c.LocalHost),
+			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, webRoutePrefix),
```
Comment (metalmatze) [inline pkg/prometheus/statefulset.go]: I suggest to handle `webRoutePrefix` similar to other parts in this function, where we use `path.Clean`.
```suggestion
			fmt.Sprintf("--prometheus.url=http://%s:9090%s", c.LocalHost, path.Clean(webRoutePrefix)),
```

https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/statefulset.go#L429

File: pkg/prometheus/statefulset.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 	"strings"
 
 	appsv1 "k8s.io/api/apps/v1beta2"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (metalmatze) [inline pkg/prometheus/statefulset.go]:
```suggestion
	"k8s.io/api/core/v1"
```

Comment (ephur) [reply pkg/prometheus/statefulset.go]: Not sure why vscode keeps trying to make this change when go fmt does not, reverted!

**Comment by metalmatze** [comment]
Thanks!

**Pull request merged**